File: app/src/main/java/fr/free/nrw/commons/bookmarks/items/BookmarkItemsController.java
package fr.free.nrw.commons.bookmarks.items;

import fr.free.nrw.commons.upload.models.depictions.DepictedItem;
import java.util.List;
import javax.inject.Inject;
import javax.inject.Singleton;

/**
 * Handles loading bookmarked items from Database
 */
@Singleton
public class BookmarkItemsController {

    @Inject
    BookmarkItemsDao bookmarkItemsDao;

    @Inject
    public BookmarkItemsController() {}

    /**
     * Load from DB the bookmarked items
     * @return a list of DepictedItem objects.
     */
    public List<DepictedItem> loadFavoritesItems() {
        return bookmarkItemsDao.getAllBookmarksItems();
    }
}

File: app/src/main/java/fr/free/nrw/commons/bookmarks/items/BookmarkItemsController.kt
package fr.free.nrw.commons.bookmarks.items

import fr.free.nrw.commons.upload.structure.depictions.DepictedItem
import javax.inject.Inject
import javax.inject.Singleton

/**
 * Handles loading bookmarked items from Database
 */
@Singleton
class BookmarkItemsController @Inject constructor() {
    @JvmField
    @Inject
    var bookmarkItemsDao: BookmarkItemsDao? = null

    /**
     * Load from DB the bookmarked items
     * @return a list of DepictedItem objects.
     */
    fun loadFavoritesItems(): List<DepictedItem> {
        return bookmarkItemsDao?.getAllBookmarksItems() ?: emptyList()
    }
}
